arch/sim/src/sim/up_idle.c

```diff
@@ -59,9 +59,13 @@ void up_idle(void)
 #ifdef CONFIG_PM
   static enum pm_state_e state = PM_NORMAL;
   enum pm_state_e newstate;
+  irqstate_t flags;
 
   /* Fake some power management stuff for testing purposes */
 
+  flags = enter_critical_section();
+  sched_lock();
```

Comment:
why do we need both critical section and scheduler lock?

Reply:
The critical section can't block the high priority thread which is made ready in the driver pm callback. Here is the related issue: https://github.com/apache/incubator-nuttx/issues/1138